core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -403,7 +401,30 @@ class LogManager(logDirs: Seq[File],
   /**
    *  Start the background threads to flush logs and do log cleanup
    */
-  def startup(): Unit = {
+  def startup(retrieveTopicNames: => Set[String]): Unit = {
+    startupWithTopicLogConfigOverrides(generateTopicLogConfigs(retrieveTopicNames))
```

Comment:
We are able to load the topic configs right away because they are coming from ZK, right?

Reply:
Yes, `startup()` is invoked immediately after the instance is constructed when we are using ZooKeeper, and we can get the configs at that point since we can just query ZK.